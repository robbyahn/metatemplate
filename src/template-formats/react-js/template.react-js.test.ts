import { testFormat } from "../test-utils";

describe("React JS", () => {
  testFormat("react-js");
});
