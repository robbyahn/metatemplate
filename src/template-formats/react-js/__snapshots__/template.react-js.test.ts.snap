// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React JS Snapshots makeIndexImports 1`] = `
Object {
  "react-ts/index.tsx": "// DEVELOPER NOTE: These components are ready to lazy-load. You may also import components directly.

export const blah.js = () => import(\\"./blah.js\\");
",
  "react-ts/indexNotLazy.tsx": "// DEVELOPER NOTE: This file includes all components so importing this file may be very inefficient. Use carefully! You may prefer to import components directly or use index.js which has import functions, typically treated as a code splitting point.

export { default as blahJs } from \\"./blah.js\\";
",
}
`;

exports[`React JS Snapshots react-js: <img src="pic.jpeg" class="frogs"> 1`] = `
Object {
  "disposeAll": [Function],
  "metaTemplates": Array [
    Object {
      "files": Object {
        "react-js/imgSrc.js": "import React from \\"react\\";

const constants = {
  crossOrigin: { Anonymous: \\"anonymous\\", \\"Use Credentials\\": \\"use-credentials\\" }
};

const ImgSrc = ({ src, width, height, srcSet, crossOrigin }) => (
  <img
    className=\\"frogs\\"
    src={src}
    width={width}
    height={height}
    srcSet={srcSet}
    crossOrigin={constants.crossOrigin[crossOrigin]}
  />
);

export default ImgSrc;
",
      },
      "formatId": "react-js",
    },
  ],
}
`;

exports[`React JS Snapshots react-js: Basic variables 1`] = `
Object {
  "disposeAll": [Function],
  "metaTemplates": Array [
    Object {
      "files": Object {
        "react-js/paragraph.js": "import React from \\"react\\";

const Paragraph = ({ children }) => (
  <p>
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Some placeholder text</React.Fragment>
    )}
  </p>
);

export default Paragraph;
",
      },
      "formatId": "react-js",
    },
  ],
}
`;

exports[`React JS Snapshots react-js: Boolean attribute values 1`] = `
Array [
  Object {
    "files": Object {
      "react-js/row.js": "import React from \\"react\\";

const Row = ({ isReversed, children }) => (
  <div className={\`g-flex-row\${isReversed ? \\" g-flex-reverse\\" : \\"\\"}\`}>
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Columns</React.Fragment>
    )}
  </div>
);

export default Row;
",
    },
    "formatId": "react-js",
  },
]
`;

exports[`React JS Snapshots react-js: Enum attribute values 1`] = `
Array [
  Object {
    "files": Object {
      "react-js/container.js": "import React from \\"react\\";

const constants = {
  containerType: {
    \\"g-flex-container\\": \\"g-flex-container\\",
    \\"g-flex-container-fluid\\": \\"g-flex-container-fluid\\"
  }
};

const Container = ({ containerType, children }) => (
  <div
    className={
      constants.containerType[containerType] !== undefined
        ? constants.containerType[containerType]
        : \\"\\"
    }
  >
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Rows...</React.Fragment>
    )}
  </div>
);

export default Container;
",
    },
    "formatId": "react-js",
  },
]
`;

exports[`React JS Snapshots react-js: Form input[checked] 1`] = `
Object {
  "disposeAll": [Function],
  "metaTemplates": Array [
    Object {
      "files": Object {
        "react-js/input.js": "import React from \\"react\\";

const Input = ({
  textId,
  name,
  disabled,
  readOnly,
  autoFocus,
  value,
  checked,
  onChange
}) => (
  <input
    id={textId}
    type=\\"checkbox\\"
    name={name}
    disabled={disabled}
    readOnly={readOnly}
    autoFocus={autoFocus}
    value={value}
    checked={checked}
    onChange={onChange}
  />
);

export default Input;
",
      },
      "formatId": "react-js",
    },
  ],
}
`;

exports[`React JS Snapshots react-js: Form label input[checked] 1`] = `
Object {
  "disposeAll": [Function],
  "metaTemplates": Array [
    Object {
      "files": Object {
        "react-js/input.js": "import React from \\"react\\";

const Input = ({
  textId,
  name,
  disabled,
  readOnly,
  autoFocus,
  value,
  checked,
  onChange
}) => (
  <label htmlFor={textId}>
    <input
      id={textId}
      type=\\"checkbox\\"
      name={name}
      disabled={disabled}
      readOnly={readOnly}
      autoFocus={autoFocus}
      value={value}
      checked={checked}
      onChange={onChange}
    />
  </label>
);

export default Input;
",
      },
      "formatId": "react-js",
    },
  ],
}
`;

exports[`React JS Snapshots react-js: H1 with configurable Id 1`] = `
Object {
  "disposeAll": [Function],
  "metaTemplates": Array [
    Object {
      "files": Object {
        "react-js/h1Id.js": "import React from \\"react\\";

const H1Id = ({ id }) => <h1 id={id}>stuff</h1>;

export default H1Id;
",
      },
      "formatId": "react-js",
    },
  ],
}
`;

exports[`React JS Snapshots react-js: Link with boolean class 1`] = `
Array [
  Object {
    "files": Object {
      "react-js/a.js": "import React from \\"react\\";

const constants = {
  target: { Blank: \\"_blank\\", Top: \\"_top\\", Self: \\"_self\\", Parent: \\"_parent\\" }
};

const A = ({ isMuted, href, rel, target, children }) => (
  <a
    className={\`g-link\${isMuted ? \\" g-link--muted\\" : \\"\\"}\`}
    href={href}
    rel={rel}
    target={constants.target[target]}
  >
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Example text</React.Fragment>
    )}
  </a>
);

export default A;
",
    },
    "formatId": "react-js",
  },
]
`;

exports[`React JS Snapshots react-js: Lots of enum attribute values 1`] = `
Array [
  Object {
    "files": Object {
      "react-js/container.js": "import React from \\"react\\";

const constants = {
  columnXs: {
    \\"g-flex-col-xs\\": \\"g-flex-col-xs\\",
    \\"g-flex-col-xs-1\\": \\"g-flex-col-xs-1\\",
    \\"g-flex-col-xs-2\\": \\"g-flex-col-xs-2\\",
    \\"g-flex-col-xs-3\\": \\"g-flex-col-xs-3\\",
    \\"g-flex-col-xs-4\\": \\"g-flex-col-xs-4\\",
    \\"g-flex-col-xs-5\\": \\"g-flex-col-xs-5\\",
    \\"g-flex-col-xs-6\\": \\"g-flex-col-xs-6\\",
    \\"g-flex-col-xs-7\\": \\"g-flex-col-xs-7\\",
    \\"g-flex-col-xs-8\\": \\"g-flex-col-xs-8\\",
    \\"g-flex-col-xs-9\\": \\"g-flex-col-xs-9\\",
    \\"g-flex-col-xs-10\\": \\"g-flex-col-xs-10\\",
    \\"g-flex-col-xs-11\\": \\"g-flex-col-xs-11\\",
    \\"g-flex-col-xs-12\\": \\"g-flex-col-xs-12\\"
  }
};

const Container = ({ isReversed, columnXs, children }) => (
  <div
    className={\`g-flex-col\${isReversed ? \\" g-flex-reverse\\" : \\"\\"}\${
      constants.columnXs[columnXs] !== undefined
        ? \\" \\" + constants.columnXs[columnXs]
        : \\"\\"
    }\`}
  >
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Columns</React.Fragment>
    )}
  </div>
);

export default Container;
",
    },
    "formatId": "react-js",
  },
]
`;

exports[`React JS Snapshots react-js: form input[text] 1`] = `
Object {
  "disposeAll": [Function],
  "metaTemplates": Array [
    Object {
      "files": Object {
        "react-js/input.js": "import React from \\"react\\";

const constants = {
  autoComplete: {
    Off: \\"off\\",
    On: \\"on\\",
    Name: \\"name\\",
    \\"Honorific: Prefix\\": \\"honorific-prefix\\",
    \\"Given Name\\": \\"given-name\\",
    \\"Additional Name\\": \\"additional-name\\",
    \\"Family Name\\": \\"family-name\\",
    \\"Honorific: Suffix\\": \\"honorific-suffix\\",
    Nickname: \\"nickname\\",
    Email: \\"email\\",
    Username: \\"username\\",
    \\"New Password\\": \\"new-password\\",
    \\"Current Password\\": \\"current-password\\",
    \\"Organization Title\\": \\"organization-title\\",
    Organization: \\"organization\\",
    \\"Street Address\\": \\"street-address\\",
    \\"Address Line 1\\": \\"address-line1\\",
    \\"Address Line 2\\": \\"address-line2\\",
    \\"Address Line 3\\": \\"address-line3\\",
    \\"Address Level 4\\": \\"address-level4\\",
    \\"Address Level 3\\": \\"address-level3\\",
    \\"Address Level 2\\": \\"address-level2\\",
    \\"Address Level 1\\": \\"address-level1\\",
    Country: \\"country\\",
    \\"Country Name\\": \\"country-name\\",
    \\"Postal Code\\": \\"postal-code\\",
    \\"Credit Card: Name\\": \\"cc-name\\",
    \\"Credit Card: Given Name\\": \\"cc-given-name\\",
    \\"Credit Card: Additional Name\\": \\"cc-additional-name\\",
    \\"Credit Card: Family Name\\": \\"cc-family-name\\",
    \\"Credit Card: Number\\": \\"cc-number\\",
    \\"Credit Card: Expiry\\": \\"cc-exp\\",
    \\"Credit Card: Expiry Month\\": \\"cc-exp-month\\",
    \\"Credit Card: Expiry Year\\": \\"cc-exp-year\\",
    \\"Credit Card: CSC\\": \\"cc-csc\\",
    \\"Credit Card: Type\\": \\"cc-type\\",
    \\"Transaction: Currency\\": \\"transaction-currency\\",
    \\"Transaction: Amount\\": \\"transaction-amount\\",
    Language: \\"language\\",
    Birthday: \\"bday\\",
    \\"Birthday: Day\\": \\"bday-day\\",
    \\"Birthday: Month\\": \\"bday-month\\",
    \\"Birthday: Year\\": \\"bday-year\\",
    Sex: \\"sex\\",
    Telephone: \\"tel\\",
    \\"Telephone: Country Code\\": \\"tel-country-code\\",
    \\"Telephone: National\\": \\"tel-national\\",
    \\"Telephone: Area Code\\": \\"tel-area-code\\",
    \\"Telephone: Local\\": \\"tel-local\\",
    \\"Telephone: Extension\\": \\"tel-extension\\",
    IMPP: \\"impp\\",
    URL: \\"url\\",
    Photo: \\"photo\\"
  }
};

const Input = ({
  textId,
  name,
  disabled,
  readOnly,
  autoFocus,
  value,
  spellCheck,
  maxLength,
  autoComplete,
  onChange
}) => (
  <React.Fragment>
    <label htmlFor={textId}>Label text</label>
    <input
      id={textId}
      type=\\"text\\"
      name={name}
      disabled={disabled}
      readOnly={readOnly}
      autoFocus={autoFocus}
      value={value}
      spellCheck={spellCheck}
      maxLength={maxLength}
      autoComplete={constants.autoComplete[autoComplete]}
      onChange={onChange}
    />
  </React.Fragment>
);

export default Input;
",
      },
      "formatId": "react-js",
    },
  ],
}
`;

exports[`React JS react-js: CSS Configuration (CSS Variables and Scss Variables) 1`] = `
Array [
  Object {
    "files": Object {
      "react-js/a.js": "import React from \\"react\\";

const constants = {
  target: { Blank: \\"_blank\\", Top: \\"_top\\", Self: \\"_self\\", Parent: \\"_parent\\" }
};

const A = ({ isMuted, href, rel, target, children }) => (
  <a
    className={\`g-link\${isMuted ? \\" g-link--muted\\" : \\"\\"}\`}
    href={href}
    rel={rel}
    target={constants.target[target]}
  >
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Example text</React.Fragment>
    )}
  </a>
);

export default A;
",
    },
    "formatId": "react-js",
  },
]
`;

exports[`React JS react-js: CSS Configuration (CSS Variables and Scss Variables) 2`] = `
Array [
  Object {
    "files": Object {
      "react-js/a.js": "import React from \\"react\\";

const constants = {
  target: { Blank: \\"_blank\\", Top: \\"_top\\", Self: \\"_self\\", Parent: \\"_parent\\" }
};

const A = ({ isMuted, href, rel, target, children }) => (
  <a
    className={\`g-link\${isMuted ? \\" g-link--muted\\" : \\"\\"}\`}
    href={href}
    rel={rel}
    target={constants.target[target]}
  >
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Example text</React.Fragment>
    )}
  </a>
);

export default A;
",
    },
    "formatId": "react-js",
  },
]
`;

exports[`React JS react-js: CSS Configuration of Fonts 1`] = `
Array [
  Object {
    "files": Object {
      "react-js/a.js": "import React from \\"react\\";

const constants = {
  target: { Blank: \\"_blank\\", Top: \\"_top\\", Self: \\"_self\\", Parent: \\"_parent\\" }
};

const A = ({ href, rel, target, children }) => (
  <a className=\\"g-link\\" href={href} rel={rel} target={constants.target[target]}>
    {children !== undefined ? (
      children
    ) : (
      <React.Fragment>Example text</React.Fragment>
    )}
  </a>
);

export default A;
",
    },
    "formatId": "react-js",
  },
]
`;

exports[`React JS react-js: CalculatedProps 1`] = `
Object {
  "disposeAll": [Function],
  "metaTemplates": Array [
    Object {
      "files": Object {
        "react-js/textbox.js": "import React from \\"react\\";

const constants = {
  type: {
    Button: \\"button\\",
    Checkbox: \\"checkbox\\",
    Color: \\"color\\",
    Date: \\"date\\",
    \\"DateTime: Local\\": \\"datetime-local\\",
    Email: \\"email\\",
    File: \\"file\\",
    Hidden: \\"hidden\\",
    Image: \\"image\\",
    Month: \\"month\\",
    Number: \\"number\\",
    Password: \\"password\\",
    Radio: \\"radio\\",
    Range: \\"range\\",
    Reset: \\"reset\\",
    Search: \\"search\\",
    Submit: \\"submit\\",
    Telephone: \\"tel\\",
    Text: \\"text\\",
    Time: \\"time\\",
    URL: \\"url\\",
    Week: \\"week\\"
  },
  autoComplete: {
    Off: \\"off\\",
    On: \\"on\\",
    Name: \\"name\\",
    \\"Honorific: Prefix\\": \\"honorific-prefix\\",
    \\"Given Name\\": \\"given-name\\",
    \\"Additional Name\\": \\"additional-name\\",
    \\"Family Name\\": \\"family-name\\",
    \\"Honorific: Suffix\\": \\"honorific-suffix\\",
    Nickname: \\"nickname\\",
    Email: \\"email\\",
    Username: \\"username\\",
    \\"New Password\\": \\"new-password\\",
    \\"Current Password\\": \\"current-password\\",
    \\"Organization Title\\": \\"organization-title\\",
    Organization: \\"organization\\",
    \\"Street Address\\": \\"street-address\\",
    \\"Address Line 1\\": \\"address-line1\\",
    \\"Address Line 2\\": \\"address-line2\\",
    \\"Address Line 3\\": \\"address-line3\\",
    \\"Address Level 4\\": \\"address-level4\\",
    \\"Address Level 3\\": \\"address-level3\\",
    \\"Address Level 2\\": \\"address-level2\\",
    \\"Address Level 1\\": \\"address-level1\\",
    Country: \\"country\\",
    \\"Country Name\\": \\"country-name\\",
    \\"Postal Code\\": \\"postal-code\\",
    \\"Credit Card: Name\\": \\"cc-name\\",
    \\"Credit Card: Given Name\\": \\"cc-given-name\\",
    \\"Credit Card: Additional Name\\": \\"cc-additional-name\\",
    \\"Credit Card: Family Name\\": \\"cc-family-name\\",
    \\"Credit Card: Number\\": \\"cc-number\\",
    \\"Credit Card: Expiry\\": \\"cc-exp\\",
    \\"Credit Card: Expiry Month\\": \\"cc-exp-month\\",
    \\"Credit Card: Expiry Year\\": \\"cc-exp-year\\",
    \\"Credit Card: CSC\\": \\"cc-csc\\",
    \\"Credit Card: Type\\": \\"cc-type\\",
    \\"Transaction: Currency\\": \\"transaction-currency\\",
    \\"Transaction: Amount\\": \\"transaction-amount\\",
    Language: \\"language\\",
    Birthday: \\"bday\\",
    \\"Birthday: Day\\": \\"bday-day\\",
    \\"Birthday: Month\\": \\"bday-month\\",
    \\"Birthday: Year\\": \\"bday-year\\",
    Sex: \\"sex\\",
    Telephone: \\"tel\\",
    \\"Telephone: Country Code\\": \\"tel-country-code\\",
    \\"Telephone: National\\": \\"tel-national\\",
    \\"Telephone: Area Code\\": \\"tel-area-code\\",
    \\"Telephone: Local\\": \\"tel-local\\",
    \\"Telephone: Extension\\": \\"tel-extension\\",
    IMPP: \\"impp\\",
    URL: \\"url\\",
    Photo: \\"photo\\"
  }
};

const Textbox = ({
  name,
  disabled,
  readOnly,
  autoFocus,
  value,
  type,
  spellCheck,
  maxLength,
  autoComplete,
  onChange
}) => (
  <input
    name={name}
    disabled={disabled}
    readOnly={readOnly}
    autoFocus={autoFocus}
    value={value}
    type={constants.type[type]}
    spellCheck={spellCheck}
    maxLength={maxLength}
    autoComplete={constants.autoComplete[autoComplete]}
    onChange={onChange}
  />
);

const Textbox__calculated = props =>
  React.createElement(Textbox, {
    ...props,
    textValueLength: (props => {
      return textValue.length;
    })(props)
  });

export { Textbox };

export default Textbox__calculated;
",
      },
      "formatId": "react-js",
    },
  ],
}
`;

exports[`React JS react-js: jsxToUsageCode with boolean false 1`] = `
Object {
  "react-js": "import React from \\"react\\";
import Ul from \\"@govtnz/ds/build/react-js/Ul\\";
import \\"@govtnz/ds/build/css/Ul.css\\"; // or @govtnz/ds/build/scss/Ul.scss
import Li from \\"@govtnz/ds/build/react-js/Li\\";
import \\"@govtnz/ds/build/css/Li.css\\"; // or @govtnz/ds/build/scss/Li.scss

export default () => (
  <Ul isBulleted=\\"false\\">
    <Li>Unordered item one</Li>
    <Li>Unordered item two</Li>
  </Ul>
);
",
}
`;

exports[`React JS react-js: jsxToUsageCode with boolean false and inline HTML 1`] = `
Object {
  "react-js": "import React from \\"react\\";
import Ul from \\"@govtnz/ds/build/react-js/Ul\\";
import \\"@govtnz/ds/build/css/Ul.css\\"; // or @govtnz/ds/build/scss/Ul.scss
import Li from \\"@govtnz/ds/build/react-js/Li\\";
import \\"@govtnz/ds/build/css/Li.css\\"; // or @govtnz/ds/build/scss/Li.scss

export default () => (
  <Ul isBulleted=\\"false\\">
    <Li>
      Unordered<a href=\\"http://zombo.com/\\">item one</a>
    </Li>
    <Li>Unordered item two</Li>
  </Ul>
);
",
}
`;

exports[`React JS react-js: jsxToUsageCode with boolean true 1`] = `
Object {
  "react-js": "import React from \\"react\\";
import Ul from \\"@govtnz/ds/build/react-js/Ul\\";
import \\"@govtnz/ds/build/css/Ul.css\\"; // or @govtnz/ds/build/scss/Ul.scss
import Li from \\"@govtnz/ds/build/react-js/Li\\";
import \\"@govtnz/ds/build/css/Li.css\\"; // or @govtnz/ds/build/scss/Li.scss

export default () => (
  <Ul isBulleted>
    <Li>Unordered item one</Li>
    <Li>Unordered item two</Li>
  </Ul>
);
",
}
`;

exports[`React JS react-js: jsxToUsageCode with enumerations 1`] = `
Object {
  "react-js": "import React from \\"react\\";
import Div from \\"@govtnz/ds/build/react-js/Div\\";
import \\"@govtnz/ds/build/css/Div.css\\"; // or @govtnz/ds/build/scss/Div.scss

export default () => <Div containerType=\\"fixed\\" />;
",
}
`;

exports[`React JS react-js: makeUsage 1`] = `
Object {
  "react-js": "import React from \\"react\\";
import H1 from \\"@govtnz/ds/build/react-js/H1\\";
import \\"@govtnz/ds/build/css/H1.css\\"; // or @govtnz/ds/build/scss/H1.scss
import A from \\"@govtnz/ds/build/react-js/A\\";
import \\"@govtnz/ds/build/css/A.css\\"; // or @govtnz/ds/build/scss/A.scss

export default () => (
  <H1>
    <A href=\\"https://html5zombo.com/\\" reversed=\\"On\\">
      Click me
    </A>
  </H1>
);
",
}
`;

exports[`React JS react-js: makeUsage with booleans 1`] = `
Object {
  "react-js": "import React from \\"react\\";
import H1 from \\"@govtnz/ds/build/react-js/H1\\";
import \\"@govtnz/ds/build/css/H1.css\\"; // or @govtnz/ds/build/scss/H1.scss
import A from \\"@govtnz/ds/build/react-js/A\\";
import \\"@govtnz/ds/build/css/A.css\\"; // or @govtnz/ds/build/scss/A.scss

export default () => (
  <H1>
    <A href=\\"https://html5zombo.com/\\" isReversed>
      Click me
    </A>
  </H1>
);
",
}
`;
